mesonbuild/interpreter/interpreter.py

```diff
@@ -1087,21 +1097,18 @@ def func_get_option(self, node: mparser.BaseNode, args: T.Tuple[str],
             ocopy.value = value
             return ocopy
         elif optname == 'b_sanitize':
-            assert isinstance(value_object, options.UserStringArrayOption)
+            assert value_object is None or isinstance(value_object, options.UserStringArrayOption)
             # To ensure backwards compatibility this always returns a string.
             # We may eventually want to introduce a new "format" kwarg that
             # allows the user to modify this behaviour, but for now this is
             # likely good enough for most usecases.
             if not value:
                 return 'none'
             return ','.join(sorted(value))
-        elif isinstance(value_object, options.UserOption):
-            if isinstance(value_object.value, str):
-                return P_OBJ.OptionString(value, f'{{{optname}}}')
-            return value
-        ocopy = copy.copy(value_object)
-        ocopy.value = value
-        return ocopy
```

Comment:
I'm not certain what you mean by:
> I've also removed some dead code here, as there was no way we could hit the clause after the if/elif/elif block except in a newly created case, which would now be an error.

W.r.t. a "newly created case". But UserOption is the base class for all options so it would rather have to be one, right?

I have the feeling you're saying that we had options which weren't options?

Reply:
the `ocopy = ...` code never be reached, since the we would always have a `UserOption` of some kind. With the change I made it could, theoretically get there with a `value_object` of None, but we don't want that.

Reply:
I split the dead code removal into a separate commit, since that's better form anyway, and will hopefully make the reason for it's removal make more sense.

Reply:
Thanks. :)

Reply:
(We do get to that if statement when value_object is None, that is the entire point of your later commit.)